src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/XUnit/XUnitTestInvokerRunAsyncIntegration.cs

```diff
@@ -0,0 +1,178 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Datadog.Trace.Ci;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.ExtensionMethods;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.XUnit
+{
+    /// <summary>
+    /// Xunit.Sdk.TestInvoker`1.RunAsync calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        Assemblies = new[] { "xunit.execution.dotnet", "xunit.execution.desktop" },
+        Type = "Xunit.Sdk.TestInvoker`1",
+        Method = "RunAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1<System.Decimal>",
+        ParametersTypesNames = new string[0],
+        MinimumVersion = "2.2.0",
+        MaximumVersion = "2.*.*",
+        IntegrationName = IntegrationName)]
+    public static class XUnitTestInvokerRunAsyncIntegration
+    {
+        private const string IntegrationName = "XUnit";
+
+        static XUnitTestInvokerRunAsyncIntegration()
+        {
+            // Preload environment variables.
+            CIEnvironmentValues.DecorateSpan(null);
+        }
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget>(TTarget instance)
+        {
+            if (!Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName))
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            TestInvokerStruct invokerInstance = instance.As<TestInvokerStruct>();
+
+            string testSuite = invokerInstance.TestClass.ToString();
```

Comment:
This shares a lot of code with TestRunnerRunAsyncIntegration with the only difference being the instance type. This code probably won't change much, so maybe not worth refactoring, but what do you think of creating a helper method that decorates a span based on all of the shared fields for example: `public void DecorateSpan(Span span, Type testClass, MethodInfo testMethod, object[] testMethodArguments, Dictionary<string, List<string>> traits)`

Reply:
Sure I will refactor it.

Reply:
I changed in 0a4ced6 using the already created `TestRunnerStruct` and passing by ref so we avoid all the copy thing.